### DISCUSSION TITLE
Rules and Guidelines as downloadable .md markdown files?

### DISCUSSION THREAD

**Discussion by SirWumpus** [ioccc-src/winner#27] [Q&A] [answered]
For each year I enter the contest, I like to save a copy of the Rules and Guidelines for local consultation (handy while travelling).  In the past they have been .txt text files.  

Will the Rules and Guidelines be available as .md Markdown files (fancier text files)?

**Comment by SirWumpus** [comment, marked as answer]
I finally found copies of the Rules & Guidelines as .md files in the [next/](https://github.com/ioccc-src/winner/tree/master/next) of the repository.  Links in the web site documentation (Rules & Guidelines) to these specific files would be nice, but not a huge deal given they do exist (`Existence is futile. Prepare to be philosophised.`) for those who know [How Not To Be Seen](https://www.youtube.com/watch?v=ifmRgQX82O4).

**Reply by xexyl** [reply]
Actually if you click on the menu called Status you will see them. 

Does that help?

**Reply by SirWumpus** [reply]
Not really. The menu only leads you to the online versions, whereas my OP was concerning downloadable Markdown files for offline viewing.

**Reply by xexyl** [reply]
> Not really. The menu only leads you to the online versions, whereas my OP was concerning downloadable Markdown files for offline viewing.

Well remember this is the first time we even have markdown files for the rules and guidelines. And that's to generate the html files only. One could certainly download the markdown files if they want though they won't see the stressing of words as well nor will they get to click on links in the same way.

For instance:

```markdown
- <p class="leftbar">[view example Makefile](%%REPO_URL%%/next/Makefile.example)</p>
```

Some links are relative but they usually link to html files which are not really useful to open in a text editor. Plus it wouldn't be offline unless you clone the repo (or do download the entire repo or a section of it I guess).

**Reply by SirWumpus** [reply]
> Well remember this is the first time we even have markdown files for the rules and guidelines. 

Well the IOCCC used to have .txt files for the Rules & Guidelines AND Markdown files by definition are suppose to be readable as text and easy to translate into HTML.  From https://daringfireball.net/projects/markdown/

> The overriding design goal for Markdown’s formatting syntax is to make it as readable as possible. The idea is that a Markdown-formatted document should be publishable as-is, as plain text ...

**Comment by lcn2** [comment]
> I finally found copies of the Rules & Guidelines as .md files in the [next/](https://github.com/ioccc-src/winner/tree/master/next) of the repository.  Links in the web site documentation (Rules & Guidelines) to these specific files would be nice, but not a huge deal given they do exist (`Existence is futile. Prepare to be philosophised.`) for those who know [How Not To Be Seen](https://www.youtube.com/watch?v=ifmRgQX82O4).

Well python quoted @SirWumpus 🤓

Other than for an inventory of files for a winning entry, links to something like `rules.md` are a bug 🪳 not a feature 🤵🏽‍♂️, as they should link to something like `rules.html` instead.

# UPDATE 0

The parallel markdown files that generate HTML will be part of the repo. But unless we are creating an inventory list of files (for say an entry), web links to the content should reference the generated HTML.  There are a few exceptions, such as say an explicit reference to some markdown example, but by and large web links should go to their HTML counterparts.

**Reply by SirWumpus** [reply]
> Other than for an inventory of files for a winning entry, links to something like rules.md are a bug 🪳 not a feature 

I'd still argue that .md files, being readable text files, should be downloaded via a link (I like to have them since I archive them with corresponding entries I made).  But maybe let other contestants weigh in as to whether or not they miss the .txt versions.

**Reply by xexyl** [reply]
> > Other than for an inventory of files for a winning entry, links to something like rules.md are a bug 🪳 not a feature
> 
> I'd still argue that .md files, being readable text files, should be downloaded via a link (I like to have them since I archive them with corresponding entries I made). But maybe let other contestants weigh in as to whether or not they miss the .txt versions.

The problem with that is that it means you have to maintain the same text in more than one place and worse is that yo would have to have different formatting. Also the link forms like I showed you is a problem. It would be a mare to deal with. Or if not a mare it will at least be a burden.

Maybe you could write a submission that converts markdown to text ? :-)

As for me I am quite fine with the html version. Makes it nicer to read: though I don't, I admit, like the white background as that hurts my eyes more.

**Reply by SirWumpus** [reply]
I thought you generate the HTMl from the Markdown,so you should only need one copy.

> Maybe you could write a submission that converts markdown to text ? :-)

No no no.  .md is a good thing over .txt files, but I should be able to get the R&G in .md format since it is a human readable text file.  I prefer Markdown since it should be nice I pretty to read in an editor too.  And I can read Markdown in Firefox using [Markdown Viewer](https://github.com/simov/markdown-viewer) (Chrome probably has similar).

> ...  I don't, I admit, like the white background as that hurts my eyes more.

Get the `Dark Reader` plugin for Chrome or Firefox.

**Reply by xexyl** [reply]
> I thought you generate the HTMl from the Markdown,so you should only need one copy.
> 

We do. I misunderstood it seems. See below.

> > Maybe you could write a submission that converts markdown to text ? :-)
> 
> No no no. .md is a good thing over .txt files, but I should be able to get the R&G in .md format since it is a human readable text file. I prefer Markdown since it should be nice I pretty to read in an editor too. And I can read Markdown in Firefox using [Markdown Viewer](https://github.com/simov/markdown-viewer) (Chrome probably has similar).
> 

Then I misunderstood. You wondered if others missed the text files: I guessed you meant text as in a text file not a file that is of text but formatted as markdown. 

You're right then. Only one copy would have to be maintained.

But why not just download them if that's your concern? You could always use wget or clone the repo or download the tarball or ...

I guess I still don't quite get your concern?


> > ...  I don't, I admit, like the white background as that hurts my eyes more.
> 
> Get the `Dark Reader` plugin for Chrome or Firefox.

Interesting. Would never use Chrome .. firefox I haven't used in ages but good to know. I mostly use Safari nowadays. But thanks for the tip!

**Reply by SirWumpus** [reply]
> I guess I still don't quite get your concern?

Well the OP was I couldn't find any reference to them in the HTML while reading the R&G this morning; not until I found them in the repo under the `next/` directory. 

Also maybe cloning a repository is over kill for some people when all they want/need are two files.  And the repositories are dense, contain lots of material to sort through, so a shortcut to the R&G in Markdown is a nice to have.  But now with this discussion people should spot my answer as to where they were spotted.

**Reply by xexyl** [reply]
> > I guess I still don't quite get your concern?
> 
> 
> 
> Well the OP was I couldn't find any reference to them in the HTML while reading the R&G this morning; not until I found them in the repo under the `next/` directory. 
> 
> 
> 
> Also maybe cloning a repository is over kill for some people when all they want/need are two files.  And the repositories are dense, contain lots of material to sort through, so a shortcut to the R&G in Markdown is a nice to have.  But now with this discussion people should spot my answer as to where they were spotted.

Ah. Well that is a fair point. And it certainly has crossed my mind. It's also a lot of megabytes. 

Maybe there could be a link somewhere or in the FAQ. But I don't know. 

What did you have in mind as a way to go about it?

**Reply by SirWumpus** [reply]
> What did you have in mind as a way to go about it?

Simply add one link to the Rules and one to the Guidelines at the top:

```
[Markdown version of this file is available.](link to corresponding .md)
```

**Comment by lcn2** [comment]
> For each year I enter the contest, I like to save a copy of the Rules and Guidelines for local consultation (handy while travelling).  In the past they have been .txt text files.  
> 
> 
> 
> Will the Rules and Guidelines be available as .md Markdown files (fancier text files)?
> 
> 

We plan to keep in 2024, the official rules and guidelines of that year, both in markdown and HTML form.  There will be an inventory that links to the markdown, but for the vast majority of web pages that need to reference the IOCCC28 rules and guidelines, they will link to the HTML page.

**Comment by lcn2** [comment]
With commit 3d9355e1c20e21b6fd2c1bad65869072c7bb4915 we have added links to allow someone to download the rules or guidelines.   For example:

<img width="534" alt="Screenshot 2024-12-29 at 23 26 28" src="https://github.com/user-attachments/assets/ae0476eb-dc2d-4247-8fc7-78e008be3ca8" />

<img width="495" alt="Screenshot 2024-12-29 at 23 27 19" src="https://github.com/user-attachments/assets/83d6f2ff-700e-4a5f-aeca-780256776e56" />

With this we believe we have resolved this comment.  Thanks @SirWumpus for your suggestion.
